src/controller/java/src/matter/controller/Messages.kt

```diff
@@ -154,7 +155,36 @@ sealed class SubscriptionState {
 class WriteRequest(
   val attributePath: AttributePath,
   val tlvPayload: ByteArray,
-)
+  val jsonString: String? = null,
+  val dataVersion: UInt? = null
+) {
+  @Suppress("UNUSED_PARAMETER")
+  fun getEndpointId(wildcardId: Long): Long {
+    return attributePath.endpointId.toLong()
```

Comment:
Use UShort for endpointId and UInt for wildcardId, we need to align with spec in kotlin

Reply:
This method is only used in JNI Layer. I will modify to private.
And, for supporting wildcard, it is necessary to change each parameter (endpointId, clusterId, attributeId) to nullable. e.g, `UShort?`.

Reply:
In the C++ code, we have the following type definitions:

typedef uint32_t AttributeId;
typedef uint32_t ClusterId;
typedef uint32_t CommandId;
typedef uint64_t CompressedFabricId;
typedef uint16_t EndpointId;
typedef uint32_t EventId;
typedef uint64_t EventNumber;
typedef uint64_t FabricId;

To align with these in Kotlin, we should ensure matching types are used in JNI.

Regarding wildcard support, we use 0xFFFFFFF to represent wildcard IDs. I don't think there's need to make them nullable.

Reply:
@yufengwangca 
In the case of a wildcard, I think null would be more appropriate because it is omited in the interaction model. (In darwin, it is expressed as nil.)